### PR TITLE
Refactor devstack local.conf and installation roles 

### PR DESCRIPTION
**Pull request by liu-sheng**
- refactor devstack local.conf to populate config according to the openstack version and request services
- refactor devstack installation to keep simple and consistent

### CODE REVIEW

File: zuul.d/jobs.yaml [hunk 1]
```diff
@@ -157,8 +138,6 @@
     description: |
```
Comment (kiwik) [inline zuul.d/jobs.yaml]: this designate job use the specific run.yaml, don't use its parents' script, so I think we can set parent as "golang-test".

File: roles/create-devstack-local-conf/tasks/main.yml [hunk 2]
```diff
@@ -50,49 +30,53 @@
...
+      enable_service designate,designate-central,designate-api,designate-pool-manager,designate-zone-manager,designate-mdns
+      enable_plugin designate git://git.openstack.org/openstack/designate
       EOF
     executable: /bin/bash
     chdir: '{{ ansible_user_dir }}/workspace'
   environment: '{{ zuul | zuul_legacy_vars }}'
   when:
-    - sdk == 'terraform-provider-openstack'
-    - os_branch in eol_branches
+    - '"designate" in enable_services'
```
Comment (kiwik) [inline roles/create-devstack-local-conf/tasks/main.yml]: see my comments in PR #20

File: roles/create-devstack-local-conf/tasks/main.yml [hunk 3]
```diff
@@ -50,49 +30,53 @@
...
+      enable_plugin heat git://git.openstack.org/openstack/heat
+      enable_plugin manila git://git.openstack.org/openstack/manila
       EOF
     executable: /bin/bash
     chdir: '{{ ansible_user_dir }}/workspace'
   environment: '{{ zuul | zuul_legacy_vars }}'
   when:
-    - sdk == 'terraform-provider-openstack'
-    - os_branch in stable_branches
+    - '"manila" in enable_services'
```
Comment (kiwik) [inline roles/create-devstack-local-conf/tasks/main.yml]: see my comments in PR #20

File: roles/create-devstack-local-conf/tasks/main.yml [hunk 4]
```diff
@@ -50,49 +30,53 @@
     chdir: '{{ ansible_user_dir }}/workspace'
   environment: '{{ zuul | zuul_legacy_vars }}'
   when:
-    - sdk == 'gophercloud'
-    - os_branch in eol_branches
+    - os_branch == 'mitaka-eol'
```
Comment (kiwik) [inline roles/create-devstack-local-conf/tasks/main.yml]: see my comments in PR #20

File: zuul.d/jobs.yaml (hunk 1 above)
Comment (animationzl) [reply zuul.d/jobs.yaml]: both 'golang-test' and 'terraform-provider-openstack-acceptance-test' as parent job will have no impact on the designate job execution however, I agree with you as choosing the 'golang-test' as parent job is more  logical.

File: roles/create-devstack-local-conf/tasks/main.yml (hunk 2 above)
Comment (animationzl) [reply roles/create-devstack-local-conf/tasks/main.yml]: Nope, this is the string in variable list syntax required by ansible, see the reference here: https://stackoverflow.com/questions/36496911/run-an-ansible-task-only-when-the-variable-contains-a-specific-string

File: roles/create-devstack-local-conf/tasks/main.yml (hunk 3 above)
Comment (animationzl) [reply roles/create-devstack-local-conf/tasks/main.yml]: ditto

**Pull request merged**